### DISCUSSION TITLE
Submitting form does full page reload / redirect

### DISCUSSION THREAD

**Discussion by AndrewMarkUK** [livewire/livewire#8983] [1. Help]
Using the logic below, the form submits, the record gets created. However, unless I am misunderstanding, because I am using wire:submit I assumed could navigate back to my index view without a full page reload. 

At the moment $this->redirectRoute('navigation.index'); (in the save method) is commented out and so when I submit the form it submits, (the record is created) and of course it sits there. If I uncomment it and look at the network tab, I see the request is processed by fetch (xhr) but, the page redirects back to the navigation.index with a full page load. Is this the expected behaviour, I followed carefully the screen cast / docs and see no additional info on getting it to show the navigation.indexc without full reload / refresh...

```
namespace App\Livewire\Admin\Site\Navigation;

use App\Models\Navigation;
use Illuminate\Contracts\View\Factory;
use Illuminate\Contracts\View\View;
use Livewire\Component;
use Illuminate\Foundation\Application;

class Create extends Component
{
    public $parentMenus;
    public $label;
    public $route;


    public function mount(): void
    {
        $this->parentMenus = Navigation::whereNull('parent_id')->get();
    }

    public function save(): void
    {
        Navigation::create([
            'label' => $this->label,
            'route' => $this->route,
        ]);

        // How does this work with wire:navigate...or does it?
       // $this->redirectRoute('navigation.index');
    }

    public function render(): Application|Factory|View|\Illuminate\View\View
    {
        return view('livewire.admin.site.navigation.create');
    }
}
```

```
<div class="container">
    <h1>Create Menu Item</h1>
    <form wire:submit="save">
        <div class="form-group mb-3">
            <label for="label">Label</label>
            <input type="text" class="form-control" id="label" name="label" wire:model="label">
        </div>
        <div class="form-group mb-3">
            <label for="route">Route</label>
            <input type="text" class="form-control" id="route" name="route" wire:model="route">
        </div>
        <button type="submit" class="btn btn-primary">Submit</button>
    </form>
</div>
```

**Comment by lrljoe** [comment]
Please:
1) Wrap your code in the code tags (three ` marks)
2) Share both your Component Code, and the relevant View/Blade code.

At the moment, nobody can provide you with "a fix", as there's nowhere near enough information to go on.

You're going to need to expand more than "the form submits", "the record is created" 
(so, the form has gone to your backend, your backend has received the data, processed it, and stored it).

Start with the basics:

1. Explain precisely what you are trying to achieve.  
a. What do you want this Livewire Component to do?
b. Is it for Authenticated or Unauthenticated users?

2. Sharing your code
Believe it or not, people here do try to help, so sharing your code allows them to do that.  If you don't want to do that, then I suggest you engage a commercial development company.
As a guide, sharing:
a. Your Component (PHP) code
b. Your view/blade code
Helps people to help you.

**Comment by AndrewMarkUK** [comment]
Thanks for that. I do apologies. I did include code from my two files already, but I neglected to enclose it so it collapsed. my bad. 

Anyways I sorted the issue. In the screencast I watched, there was no mention of setting a 'navigate: true' flag. This didn't make sense as the  wire: submit on the form was clearly submitting the request but the page was also then redirecting with a full page refresh. After some more digging around (in the repo in this case), I found 

```$this->redirect('/site/navigation', navigate: true);```

I do appreciate you taking the time to reply. Thank you.
